app/client-app/src/app/pages/Editor/components/PageList/PageHeader.js

```diff
@@ -1,12 +1,33 @@
-import { Flex, useColorMode, Icon } from "@chakra-ui/react";
+import { Flex, useColorMode, Icon, Input } from "@chakra-ui/react";
+import { useState, useEffect, useRef } from "react";
 import LightHeading from "components/core/LightHeading";
 import { FaGripVertical } from "react-icons/fa";
 import PageActionButtons from "./PageActionButtons";
 import { defaultColorMode } from "themes";
-
+import axios from "axios";
+import { useFetchSurvey } from "app/contexts/FetchSurvey";
+import { authorization_BearerToken, withHeaders, contentType_AppJson } from "api/helpers";
 const PageHeader = ({ page, order, dragHandleProps }) => {
   const { colorMode } = useColorMode();
   const headerStyle = { light: { bg: "gray.300" }, dark: { bg: "gray.600" } };
+  const { id, mutate } = useFetchSurvey();
+  const [name, setName] = useState(page.name);
+  const firstUpdate = useRef(true);
+  useEffect(() => {
+    if (firstUpdate.current == true) {
+      firstUpdate.current = false
+      return
+    }
+    const delayDebounceFn = setTimeout(async () => {
+      axios.put(
```

Comment:
this should be a function in an api file. React Components shouldn't care about how to make an API call (or even that they are); they should just care about getting/sending some data. Also, if another Component needs to make the same API call, it should be reusable.

So the axios call should go in a reusable function [here](https://github.com/decsys/decsys/blob/810ace8e54993aa0f4d4858b6eacaf80594ffa0e/app/client-app/src/api/pages.js)

This will also save importing all the low-level http helpers for headers and stuff that, again, Components shouldn't care about